sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/BatchTestBase.java

```diff
@@ -0,0 +1,185 @@
+
+import com.azure.cosmos.implementation.ISessionToken;
+import com.azure.cosmos.implementation.SessionTokenHelper;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.util.Random;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BatchTestBase extends TestSuiteBase {
+
+    private Random random = new Random();
+    String partitionKey1 = "TBD1";
+
+    // Documents in partitionKey1
+    TestDoc TestDocPk1ExistingA;
+    TestDoc TestDocPk1ExistingB ;
+    TestDoc TestDocPk1ExistingC;
+    TestDoc TestDocPk1ExistingD;
+
+    public BatchTestBase(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    void createJsonTestDocsAsync(CosmosContainer container) {
+        this.TestDocPk1ExistingA =  this.createJsonTestDocAsync(container, this.partitionKey1);
+        this.TestDocPk1ExistingB =  this.createJsonTestDocAsync(container, this.partitionKey1);
+        this.TestDocPk1ExistingC =  this.createJsonTestDocAsync(container, this.partitionKey1);
+        this.TestDocPk1ExistingD =  this.createJsonTestDocAsync(container, this.partitionKey1);
+    }
+
+    TestDoc populateTestDoc(String partitionKey) {
+        return populateTestDoc(partitionKey, 20);
+    }
+
+    TestDoc populateTestDoc(String partitionKey, int minDesiredSize) {
+        String description = StringUtils.repeat("x", minDesiredSize);
+        return new TestDoc(UUID.randomUUID().toString(), this.random.nextInt(), description, partitionKey);
+    }
+
+    TestDoc getTestDocCopy(TestDoc testDoc) {
+        return new TestDoc(testDoc.getId(), testDoc.getCost(), testDoc.getDescription(), testDoc.getStatus());
+    }
+
+    void verifyByReadAsync(CosmosContainer container, TestDoc doc) {
+        verifyByReadAsync(container, doc, null);
+    }
+
+    void verifyByReadAsync(CosmosContainer container, TestDoc doc, String eTag) {
+        PartitionKey partitionKey = this.getPartitionKey(doc.getStatus());
+
+        CosmosItemResponse<TestDoc> response = container.readItem(doc.getId(), partitionKey, TestDoc.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpResponseStatus.OK.code());
+        assertThat(response.getItem()).isEqualTo(doc);
+
+        if (eTag != null) {
+            assertThat(response.getETag()).isEqualTo(eTag);
+        }
+    }
+
+    void verifyNotFoundAsync(CosmosContainer container, TestDoc doc) {
+        String id = doc.getId();
+        PartitionKey partitionKey = this.getPartitionKey(doc.getStatus());
+
+        try {
+            CosmosItemResponse<TestDoc> response =  container.readItem(id, partitionKey, TestDoc.class);
+
+            // Gateway returns response instead of exception
```

Comment:
This shouldn't be the case.
`container.readItem()` either return a result or throw exception. 404 should never be returned as status code.

Reply:
Yeah. I tried it again, it is now throwing exception for gateway too.. must have been some mixup.